corehq/tabs/uitab.py

```diff
@@ -66,12 +66,9 @@ def request_path(self):
         return self._request.get_full_path()
 
     @property
-    def restrict_access_by_location(self):
-        """Is this a web user who can't access project-wide data?"""
-        return (
-            self.couch_user and self.domain
-            and not self.couch_user.has_permission(self.domain, 'access_all_locations')
-        )
+    def can_access_all_locations(self):
+        """Is this a web user who can access project-wide data?"""
+        return getattr(self._request, 'can_access_all_locations', True)
```

Comment:
i think if we default to True we're opening ourselves up to a vulnerability that's impossible to notice when it happens.

i'm not sure i understand the original code change well enough but is the main issue here that we just need should_show to return False, and bypass the filtered_dropdown_items and filtered_sidebar_items in this situation? if so is there anything else we can use to do that?